spirv_msl.cpp

```diff
@@ -1004,9 +1052,11 @@ void CompilerMSL::emit_fixup()
 	{
 		if (options.vertex.fixup_clipspace)
 		{
-			const char *suffix = backend.float_literal_suffix ? "f" : "";
+			/*const char *suffix = backend.float_literal_suffix ? "f" : "";
 			statement(qual_pos_var_name, ".z = 2.0", suffix, " * ", qual_pos_var_name, ".z - ", qual_pos_var_name,
-			          ".w;", "    // Adjust clip-space for Metal");
+			          ".w;", "    // Adjust clip-space for Metal");*/
+			statement(qual_pos_var_name, ".z = (", qual_pos_var_name, ".z + ", qual_pos_var_name,
```

Comment:
I hope i understood your question correctly. Metal normalized device coordinate is a cuboid [-1,1]x[-1,1]x[0,1].  For x and y the coordinates are in range between -1 and 1. But the z coordinate is in range between 0 and 1.